### PR TITLE
Asim/fix web parsers

### PR DESCRIPTION
**Pull request by oshezaf**

### CODE REVIEW

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionEmpty.yaml [hunk 1]
```diff
@@ -81,8 +81,8 @@ ParserQuery: |
   , DstMacAddr:string // Optional
   , DstGeoCountry:string // Optional
   , DstGeoCity:string // Optional
-  , DstGeoLatitude:string // Optional
-  , DstGeoLongitude:string // Optional
+  , DstGeoLatitude:real // Optional
```
Comment (yaronMSFT) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionEmpty.yaml:84]: Missing version update?

File: Parsers/ASimWebSession/Parsers/ASimWebSession.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 Parser:
   Title: Web Session ASIM parser
-  Version: '0.1'
-  LastUpdated: Nov 30, 2021
+  Version: '0.2'
```
Comment (yaronMSFT) [inline Parsers/ASimWebSession/Parsers/ASimWebSession.yaml:3]: Even though nothing changed directly on this metaparser - you update the version? 
I think we need to set rules to how and when we promote version numbers

File: Parsers/ASimWebSession/Parsers/ASimWebSessionSquidProxy.yaml [hunk 3]
```diff
@@ -53,14 +54,28 @@ ParserQuery: |
...
+    | extend 
+      DstFQDN = iif (DstIpAddrIsHost, DstIpAddr, tostring(parse_url(Url)["Host"])),
+      DstIpAddr = iif (DstIpAddr == "-" or DstIpAddrIsHost, "", DstIpAddr)
+    | extend 
+      EventSeverity = iff(EventResult == "Success", "Informational", "Low"),
+      DstFQDNparts = split (DstFQDN, ".")
+    | extend    
+      DstHostname = tostring(DstFQDNparts[0]),
+      DstDomain = strcat_array(array_slice(DstFQDNparts,1,-1),"."),
+      DstDOmainType = "FQDN"
```
Comment (yaronMSFT) [inline Parsers/ASimWebSession/Parsers/ASimWebSessionSquidProxy.yaml:68]: should be: DstDomainType

File: Parsers/ASimWebSession/Parsers/ASimWebSessionSquidProxy.yaml [hunk 4]
```diff
@@ -53,14 +54,28 @@ ParserQuery: |
       UsernameType = "Unknown",
       SrcUsername = iff (SrcUsername == "-", "", SrcUsername), 
       HttpContentType = iff (HttpContentType in (":", "-"), "", HttpContentType), 
-      EventResult = iff (EventResultOriginalDetails has_any ('DENIED', 'INVALID', 'FAIL', 'ABORTED','TIMEOUT') or EventResultDetails >= 400, "Failure", "Success")
-    // -- aliases
+      EventResult = iff (EventOriginalResultDetails has_any ('DENIED', 'INVALID', 'FAIL', 'ABORTED','TIMEOUT') or toint(EventResultDetails) >= 400, "Failure", "Success"),
+      DstIpAddrIsHost = DstIpAddr matches regex "[a-zA-Z]$"
```
Comment (yaronMSFT) [inline Parsers/ASimWebSession/Parsers/ASimWebSessionSquidProxy.yaml:58]: you are only checking the last character. is that ok?

File: Parsers/ASimWebSession/Parsers/ASimWebSessionzScalerZIA.yaml [hunk 5]
```diff
@@ -50,43 +53,48 @@ ParserQuery: |
...
+    DstHostname = iff (DstHostnameNotAddr, tostring(DstFQDNparts[0]), DstFQDN),
+    DstDomain = iff (DstHostnameNotAddr, strcat_array(array_slice(DstFQDNparts,1,-1),"."), ""),
+    DstFQDN = iff (DstHostnameNotAddr, DstFQDN, "") 
   // -- Enrichment
   | extend
     EventResult = iff (EventResultDetails == "NA" or toint(EventResultDetails) >= 400, "Failure", "Success"),
+    EventSeverity = case (ThreatRiskLevel > 90, "High", ThreatRiskLevel > 60, "Medium", ThreatRiskLevel > 10, "Low",  "Informational"),
     DstAppType = "SaaS application",
-    DstFQDN = "FQDN",
+    DstDomainType = iff (DstHostnameNotAddr, "FQDN", ""),
```
Comment (yaronMSFT) [inline Parsers/ASimWebSession/Parsers/ASimWebSessionzScalerZIA.yaml:97]: Maybe we are missing a DomainType="IP"?

File: Parsers/ASimWebSession/Parsers/imWebSession.yaml [hunk 6]
```diff
@@ -1,7 +1,7 @@
 Parser:
   Title: Web Session ASIM filtering parser
-  Version: '0.2'
-  LastUpdated: Nov 30, 2021
+  Version: '0.3'
```
Comment (yaronMSFT) [inline Parsers/ASimWebSession/Parsers/imWebSession.yaml:3]: Same comment as ASimWebSession

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionEmpty.yaml (hunk 1 above)
Comment (oshezaf) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionEmpty.yaml:84]: It accidentally got here :-(. It is part of a different PR for network fixes. Anyways, I will update the version number as part of the network update PR.

File: Parsers/ASimWebSession/Parsers/ASimWebSession.yaml (hunk 2 above)
Comment (oshezaf) [reply Parsers/ASimWebSession/Parsers/ASimWebSession.yaml:3]: Makes sense. This time I will keep it.

File: Parsers/ASimWebSession/Parsers/ASimWebSessionSquidProxy.yaml (hunk 4 above)
Comment (oshezaf) [reply Parsers/ASimWebSession/Parsers/ASimWebSessionSquidProxy.yaml:58]: Now that you mentioned that. Not for IPv6 :-). I will think of an alternative.

File: Parsers/ASimWebSession/Parsers/ASimWebSessionzScalerZIA.yaml (hunk 5 above)
Comment (oshezaf) [reply Parsers/ASimWebSession/Parsers/ASimWebSessionzScalerZIA.yaml:97]: Actually, I don't set a domain if the hostname is an IP.

File: Parsers/ASimWebSession/Parsers/ASimWebSessionSquidProxy.yaml (hunk 4 above)
Comment (oshezaf) [reply Parsers/ASimWebSession/Parsers/ASimWebSessionSquidProxy.yaml:58]: Fixed

**Comment by oshezaf** [comment]
@YaronFruchtmann : resolved. Can you check again?

**Review by yaronMSFT — APPROVED** [review]

**Review by YaronFruchtmann — APPROVED** [review]

**Pull request merged**